src/ImageFitting/InitialValueCalculator.cc

```diff
@@ -0,0 +1,329 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018- Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#define SIGMA_TO_FWHM std::sqrt(8.0 * std::log(2.0))
+
+#include "InitialValueCalculator.h"
+
+using namespace carta;
+
+InitialValueCalculator::InitialValueCalculator(FitData* fit_data, std::vector<CARTA::GaussianComponent>& initial_values, float image_std)
+    : _image(fit_data->data),
+      _width(fit_data->width),
+      _height(fit_data->n / fit_data->width),
+      _offset_x(fit_data->offset_x),
+      _offset_y(fit_data->offset_y),
+      _initial_values(initial_values),
```

Comment:
Sorry; I've realised that this is an in-out parameter, and setting it as a property is a bit awkward. I also think that the log construction can be merged into the public calculation function (since this is a log of the calculation process).

My revised suggestion:
* Don't pass `initial_values` to the constructor
* Do pass it into `CalculateInitialValues` as an in-out parameter
* Pass the image unit into the constructor instead
* Pass `initial_value_log` into `CalculateInitialValues` as another in-out parameter
* Build the log string inside `CalculateInitialValues`.